harness/determined/event_trail/event_trail_events.py

```diff
@@ -0,0 +1,69 @@
+from typing import Dict, Any
+from enum import Enum
+from abc import ABC, abstractmethod
+import json  # TODO: Simplejson instead?
+
+
+class EventPriority(Enum):
+    MUST_DELIVER = 1
+    DROPPABLE = 2
+
+
+class EventTrailEvent(ABC):
+
+
+    @property
+    @abstractmethod
+    def priority(self) -> EventPriority:
+        pass
+
+
+    @property
+    @abstractmethod
+    def version(self) -> int:
+        pass
+
+    @abstractmethod
+    def as_dict(self) -> Dict[str, Any]:
+        pass
+
+    def __str__(self):
+        return json.dumps(self.as_dict())
+
+    # TODO: Should the delivery code be included inside of the EventTrailEvent? It makes it easier
+    #       to know how to add a new event, but it makes batching sends harder.
```

Comment:
Oooh, if you are going to do batching (which I think is a good idea) you should set things up for that from the beginning, because the endpoints in particular would be very different.

At a minimum you should set up the endpoints to take lists of messages, even if for now you only send a length-1 message every time.